secp256k1-sys/vendor-libsecp.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+set -e
+
+
+if [ -z "$1" ] | [ -z "$2" ]; then
+  echo "\$1 parameter must be the rust-secp256k1-sys depend directory"
+  echo "\$2 parameter must be the rust-secp256k1-sys version code (M_m_p format)"
+  echo "\$3 parameter (optional) can be the revision to check out"
+  exit 1
+fi
+
+PARENT_DIR=$1
+VERSIONCODE=$2
+REV=$3
+DIR=secp256k1
+ORIGDIR=$(pwd)
+
+while true; do
+    read -r -p "$PARENT_DIR/$DIR will be deleted [yn]: " yn
+    case $yn in
+        [Yy]* ) break;;
+        [Nn]* ) exit;;
+        * ) echo "Please answer yes or no.";;
+    esac
+done
+
+cd "$PARENT_DIR"
+rm -rf "$DIR"
+git clone git@github.com:bitcoin-core/secp256k1.git
+cd "$DIR"
+if [ -n "$REV" ]; then
+    git checkout "$REV"
+fi
+HEAD=$(git rev-parse HEAD)
+cd ..
+echo "\# This file was automatically created by $0" > ./secp256k1-HEAD-revision.txt
+echo "$HEAD" >> ./secp256k1-HEAD-revision.txt
+
+# We need to make some source changes to the files.
+
+# We need to remove the secp_context_create and _destroy methods because we redefine them in Rust.
+patch "$DIR/include/secp256k1.h" "./secp256k1.h.patch"
+patch "$DIR/src/secp256k1.c" "./secp256k1.c.patch"
+patch "$DIR/src/scratch_impl.h" "./scratch_impl.h.patch"
+patch "$DIR/src/util.h" "./util.h.patch"
```

Comment:
Would it be simpler to rename the Rust functions?

Reply:
Maybe add a comment why we remove the other functions.

Reply:
Apparently the existence of the functions breaks a WASM build. I have not verified that, but that's why they were removed in the first place.

Reply:
Yep. It your build target has no `malloc` symbol than it will fail linking.  (which some of the wasm targets has no malloc symbols